Archival failure: invalid UTF-8

~~~~
[2020-01-07T20:44:18.820585657+01:00][sarchive][ERROR] processing failed: Custom { kind: InvalidData, error: "stream did not contain valid UTF-8" }
~~~~

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,8 +25,8 @@ crossbeam-queue = "0.2.0"
 crossbeam-utils = "0.7.0"
 fern = { version = "0.5.9", features = ["reopen-03"]}
 glob = "0.3.0"
-#notify = "5.0.0-pre.1"
-notify = { git = "https://github.com/passcod/notify", rev = "be284e3" }  # following latest main branch
+notify = "5.0.0-pre.2"
+#notify = { git = "https://github.com/passcod/notify", rev = "be284e3" }  # following latest main branch
 libc = "0.2.60"
 log = "^0.4"
 reopen = "^0.3.0"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,8 +26,7 @@ crossbeam-utils = "0.7.0"
 fern = { version = "0.5.9", features = ["reopen-03"]}
 glob = "0.3.0"
 notify = "5.0.0-pre.2"
-#notify = { git = "https://github.com/passcod/notify", rev = "be284e3" }  # following latest main branch
-libc = "0.2.60"
+libc = "0.2.66"
 log = "^0.4"
 reopen = "^0.3.0"
 signal-hook = "^0.1.12"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,7 +26,7 @@ crossbeam-utils = "0.7.0"
 fern = { version = "0.5.9", features = ["reopen-03"]}
 glob = "0.3.0"
 #notify = "5.0.0-pre.1"
-notify = { git = "https://github.com/passcod/notify", rev = "e04ec0236b2" }  # following latest main branch
+notify = { git = "https://github.com/passcod/notify", rev = "be284e3" }  # following latest main branch
 libc = "0.2.60"
 log = "^0.4"
 reopen = "^0.3.0"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "sarchive"
-version = "0.11.2"
+version = "0.11.3"
 authors = ["Andy Georges <itkovian@gmail.com>"]
 edition = "2018"
 description = "Archival tool for slurm job scripts"

--- a/src/archive/kafka.rs
+++ b/src/archive/kafka.rs
@@ -123,7 +123,7 @@ impl Archive for KafkaArchive {
                 Ok(_) => {
                     debug!("Message produced correctly");
                     Ok(())
-                },
+                }
                 Err((_e, _)) => {
                     debug!("Could not produce job entry");
                     Ok(())

--- a/src/archive/kafka.rs
+++ b/src/archive/kafka.rs
@@ -115,11 +115,15 @@ impl Archive for KafkaArchive {
         };
 
         if let Ok(serial) = serde_json::to_string(&doc) {
+            debug!("Serialisation succeeded");
             match self
                 .producer
                 .send::<str, str>(BaseRecord::to(&self.topic).payload(&serial))
             {
-                Ok(_) => Ok(()),
+                Ok(_) => {
+                    debug!("Message produced correctly");
+                    Ok(())
+                },
                 Err((_e, _)) => {
                     debug!("Could not produce job entry");
                     Ok(())

--- a/src/scheduler/slurm.rs
+++ b/src/scheduler/slurm.rs
@@ -25,8 +25,8 @@ use notify::event::{CreateKind, Event, EventKind};
 use std::collections::HashMap;
 use std::io::Error;
 use std::path::{Path, PathBuf};
-use std::time::Instant;
 use std::string::String;
+use std::time::Instant;
 
 use super::job::JobInfo;
 use super::Scheduler;
@@ -132,32 +132,29 @@ impl JobInfo for SlurmJobEntry {
     /// Returns the environment info (if any) as a HashMap, mapping env keys
     /// to values
     fn extra_info(&self) -> Option<HashMap<String, String>> {
-
-        self.env_.as_ref().map(
-            |s| {
-                let s = String::from_utf8(s.as_bytes().split_at(4).1.to_vec()).unwrap();
-                s.split('\0')
-                    .filter_map(|s| {
-                        let s = s.trim();
-                        if !s.is_empty() {
-                            let ps: Vec<_> = s.split('=').collect();
-                            match ps.len() {
-                                2 => {
-                                    if !ps[0].trim().is_empty() {
-                                        Some((ps[0].to_owned(), ps[1].to_owned()))
-                                    } else {
-                                        None
-                                    }
+        self.env_.as_ref().map(|s| {
+            let s = String::from_utf8(s.as_bytes().split_at(4).1.to_vec()).unwrap();
+            s.split('\0')
+                .filter_map(|s| {
+                    let s = s.trim();
+                    if !s.is_empty() {
+                        let ps: Vec<_> = s.split('=').collect();
+                        match ps.len() {
+                            2 => {
+                                if !ps[0].trim().is_empty() {
+                                    Some((ps[0].to_owned(), ps[1].to_owned()))
+                                } else {
+                                    None
                                 }
-                                _ => Some((s.to_owned(), String::from(""))),
                             }
-                        } else {
-                            None
+                            _ => Some((s.to_owned(), String::from(""))),
                         }
+                    } else {
+                        None
+                    }
                 })
                 .collect::<HashMap<String, String>>()
-            }
-        )
+        })
     }
 }
 

--- a/src/scheduler/slurm.rs
+++ b/src/scheduler/slurm.rs
@@ -26,6 +26,7 @@ use std::collections::HashMap;
 use std::io::Error;
 use std::path::{Path, PathBuf};
 use std::time::Instant;
+use std::string::String;
 
 use super::job::JobInfo;
 use super::Scheduler;
@@ -131,28 +132,32 @@ impl JobInfo for SlurmJobEntry {
     /// Returns the environment info (if any) as a HashMap, mapping env keys
     /// to values
     fn extra_info(&self) -> Option<HashMap<String, String>> {
-        self.env_.as_ref().map(|s| {
-            s.split('\0')
-                .filter_map(|s| {
-                    let s = s.trim();
-                    if !s.is_empty() {
-                        let ps: Vec<_> = s.split('=').collect();
-                        match ps.len() {
-                            2 => {
-                                if !ps[0].trim().is_empty() {
-                                    Some((ps[0].to_owned(), ps[1].to_owned()))
-                                } else {
-                                    None
+
+        self.env_.as_ref().map(
+            |s| {
+                let s = String::from_utf8(s.as_bytes().split_at(4).1.to_vec()).unwrap();
+                s.split('\0')
+                    .filter_map(|s| {
+                        let s = s.trim();
+                        if !s.is_empty() {
+                            let ps: Vec<_> = s.split('=').collect();
+                            match ps.len() {
+                                2 => {
+                                    if !ps[0].trim().is_empty() {
+                                        Some((ps[0].to_owned(), ps[1].to_owned()))
+                                    } else {
+                                        None
+                                    }
                                 }
+                                _ => Some((s.to_owned(), String::from(""))),
                             }
-                            _ => Some((s.to_owned(), String::from(""))),
+                        } else {
+                            None
                         }
-                    } else {
-                        None
-                    }
                 })
-                .collect()
-        })
+                .collect::<HashMap<String, String>>()
+            }
+        )
     }
 }
 

--- a/src/scheduler/slurm.rs
+++ b/src/scheduler/slurm.rs
@@ -291,7 +291,7 @@ mod tests {
         slurm_job_entry.read_job_info().unwrap();
 
         if let Some(hm) = slurm_job_entry.extra_info() {
-            assert_eq!(hm.len(), 46);
+            assert_eq!(hm.len(), 45);
             assert_eq!(hm.get("SLURM_CLUSTERS").unwrap(), "cluster");
             assert_eq!(hm.get("SLURM_NTASKS_PER_NODE").unwrap(), "1");
         } else {

